### PR TITLE
amp-sidebar - Whitelist checks and Documentation.

### PR DESCRIPTION
**Pull request by sriramkrish85**

### CODE REVIEW

File: extensions/amp-sidebar/0.1/amp-sidebar.js [hunk 1]
```diff
@@ -16,8 +16,9 @@
 
 import {CSS} from '../../../build/amp-sidebar-0.1.css';
 import {Layout} from '../../../src/layout';
-import {isExperimentOn} from '../../../src/experiments';
+import {assert} from '../../../src/asserts';
```
Comment (dvoytenko) [inline extensions/amp-sidebar/0.1/amp-sidebar.js]: Please no longer use `assers.js` - user `user from log.js` instead.

File: extensions/amp-sidebar/0.1/amp-sidebar.js [hunk 2]
```diff
@@ -185,6 +193,30 @@ export class AmpSidebar extends AMP.BaseElement {
       });
     }
   }
+
+  /**
+   * Checks if the sidebar only has the whitlisted custom amp- elements.
+   * @returns {boolean} True when only whitelited elements are present.
```
Comment (dvoytenko) [inline extensions/amp-sidebar/0.1/amp-sidebar.js]: The assert throws exceptions. Why return boolean?

File: extensions/amp-sidebar/0.1/amp-sidebar.js (hunk 1 above)
Comment (sriramkrish85) [inline extensions/amp-sidebar/0.1/amp-sidebar.js]: Done

File: extensions/amp-sidebar/0.1/amp-sidebar.js (hunk 2 above)
Comment (sriramkrish85) [inline extensions/amp-sidebar/0.1/amp-sidebar.js]: Fixed

**Comment by sriramkrish85** [comment]
PTAL

File: extensions/amp-sidebar/amp-sidebar.md [hunk 3]
```diff
@@ -23,6 +23,72 @@ limitations under the License.
   </tr>
   <tr>
     <td width="40%"><strong>Availability</strong></td>
-    <td><div>Experimental</div><div>Work in progress.</div></td>
+    <td><div><a href="https://www.ampproject.org/docs/reference/experimental.html">Experimental</a>; no validations yet.</div><div>Work in progress.</div></td>
```
Comment (dvoytenko) [inline extensions/amp-sidebar/amp-sidebar.md]: Looks like at this point we can already file the validator request, right?

Comment (dvoytenko) [inline extensions/amp-sidebar/amp-sidebar.md]: If so, please do file.

File: extensions/amp-sidebar/0.1/amp-sidebar.js [hunk 4]
```diff
@@ -234,6 +239,28 @@ export class AmpSidebar extends AMP.BaseElement {
       }
     });
   }
+
+  /**
+   * Checks if the sidebar only has the whitlisted custom amp- elements.
+   * @private
+   */
+  checkWhitelist_() {
+    const elements = this.element.getElementsByTagName('*');
```
Comment (dvoytenko) [inline extensions/amp-sidebar/0.1/amp-sidebar.js]: Please replace with `getElementsByClassName('-amp-element')`

**Comment by dvoytenko** [comment]
One significant change requested to improve scan. Otherwise, LGTM.

File: extensions/amp-sidebar/amp-sidebar.md (hunk 3 above)
Comment (sriramkrish85) [inline extensions/amp-sidebar/amp-sidebar.md]: Validator changes are already in goldmine.

Comment (sriramkrish85) [inline extensions/amp-sidebar/amp-sidebar.md]: Looks like the changes are already on goldmine :) ill have to file bugs for some minor changes.

File: extensions/amp-sidebar/0.1/amp-sidebar.js (hunk 4 above)
Comment (sriramkrish85) [inline extensions/amp-sidebar/0.1/amp-sidebar.js]: Done

**Pull request merged**